scripts/vad_overlap_posterior.py

```diff
@@ -0,0 +1,268 @@
+
+            for i, og_pred in enumerate(frame):
+                if i % jump_on_frame != 0:
+                    continue
+                start = i * shift
+                end = start + seg
+                preds[start:end] = preds[start:end] + og_pred
+                pred_count[start:end] = pred_count[start:end] + 1
+
+            preds = preds / pred_count
+            last_non_zero_pred = preds[pred_count != 0][-1]
+            preds[pred_count == 0] = last_non_zero_pred
+
+            preds = preds[:-1]
+
+        elif method == 'median':
+            preds = [[] for _ in range(target_len)]
+            for i, og_pred in enumerate(frame):
+                if i % jump_on_frame != 0:
+                    continue
+                start = i * shift
+                end = start + seg
+                for i in range(start, end):
+                    preds[i].append(og_pred)
+
+            preds = np.array([np.median(l) for l in preds])
+
+            nan_idx = np.isnan(preds)
+            last_non_nan_pred = preds[~nan_idx][-1]
+            preds[nan_idx] = last_non_nan_pred
+            preds = preds[:-1]
+
+        else:
+            raise Error("method should be either mean or median")
+
+        round_final = np.round(preds, 4)
+        np.savetxt(overlap_filepath, round_final, delimiter='\n')
+        print(f"Finished! {overlap_filepath}!")
+
+    except Exception as e:
+        with open(os.path.join(out_dir, "error.log"), "a") as f:
+            f.write(f"{name}: {e}")
+            f.write("\n")
+
+
+def gen_seg_table(frame_filepath, per_args):
+
+    """
+    Convert frame level prediction to speech/no-speech segment in start and end times format.
+    And save to csv file  in rttm-like format
+            0, 10, speech
+            10,12, no-speech
+    Args:
+        frame_filepath : frame prediction file to be processed.
+        per_args : 
+            threshold : threshold for prediction score (from 0 to 1).
+            seg_len : Length of window for generating the frame.
+            shift_len : Amount of shift of window for generating the frame.
+            has_all_frame: whether it contains frames in first half windows and very last half window.
+                           E.g. True if use frame prediction from NeMo/examples/asr/vad_infer.py;
+                                False if generate predictions with overlapping input segments.
+            out_dir : Output dir of generated table/csv file.                   
+    """
+    threshold = per_args['threshold']
+    seg_len = per_args['seg_len']
+    shift_len = per_args['shift_len']
+    has_all_frame = per_args['has_all_frame']
+    out_dir = per_args['out_dir']
+
+    print(f"process {frame_filepath}")
+    name = frame_filepath.split("/")[-1].split(".")[0]
+
+    sequence = np.loadtxt(frame_filepath)
+    first = 0
+    if not has_all_frame:
+        # if not has_all_frame indicate that lack of frames in first half windows and very last half window
```

Comment:
is this comment required?

Reply:
Nah if 'append zero for last slice' done. Will update